Aplib.Core.Tests/Belief/BeliefTests.cs

```diff
@@ -30,6 +33,117 @@ public IEnumerator<int> GetEnumerator()
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
     }
 
+    public class TestBelief : Belief<object, object>
+    {
+        public object Reference => _reference;
+
+        public Func<object, object> GetObservationFromReference => _getObservationFromReference;
+
+        public Func<bool> ShouldUpdate => _shouldUpdate;
+
+        public TestBelief
```

Comment:
If I understand correctly, you make these classes every time because you try to access protected variables.

If so, I suggest you add an internal method to Belief etc which exposes the protected variables for testing. Whilst protected fields cannot be seen by non-implementing types, internal variables can be exposed to referenced projects. Might reduce code coupling

Reply:
Wow, no, no, no. We don't add functionality just to make testing easier. That is such a bad practice.

Reply:
Code duplication or code coupling in tests is allowed, since they are just tests. In tests, it is better to define everything twice, since a test should be readable in a glance. The setup, act, and assert should be clear from one method. If we need a mock or belief to access internal variables, we do so via mocking, and not by editing the code structure.